flocker/control/_model.py

```diff
@@ -689,6 +730,57 @@ def __new__(cls, **kwargs):
     def update_cluster_state(self, cluster_state):
         return cluster_state.update_node(self)
 
+    # Attributes that may be set to None to indicate ignorance:
+    _POTENTIALLY_IGNORANT_ATTRIBUTES = ["used_ports", "applications",
+                                        "manifestations", "paths",
+                                        "devices"]
```

Comment:
Too bad these can't be actual field instances.  But it seems like field instances are really inconvenient to use as anything exception descriptors.